src/Microsoft.AspNetCore.Razor.Design/Internal/AssemblyTagHelperDescriptorResolver.cs

```diff
@@ -61,6 +52,47 @@ public IEnumerable<TagHelperDescriptor> Resolve(string assemblyName, ErrorSink e
                         DesignResources.InvalidProtocolValue,
                         typeof(TagHelperDescriptor).FullName, ProtocolVersion));
             }
+
+            if (!_loadedAssembly)
+            {
+                LoadMvcAssembly();
+            }
+
+            if (_hasMvcAssembly)
+            {
+                var serviceCollection = new ServiceCollection();
+                serviceCollection.AddSingleton<ITagHelperTypeResolver>(_tagHelperTypeResolver);
+                serviceCollection.AddSingleton<ITagHelperDescriptorFactory>(_tagHelperDescriptorFactory);
+
+                // Populate the service collection
+                _populateMethod.Invoke(_serviceCollectionProviderClassInstance, new object[] { serviceCollection, assemblyName });
```

Comment:
This runs for every assembly that we resolve descriptors from.
Example, in the following command, it runs twice.
 `dotnet razor-tooling resolve-taghelpers .\project.json ViewComponentTagHelper.Web Microsoft.AspNetCore.Mvc.TagHelpers`